keras_hub/src/models/qwen/qwen_tokenizer.py

```diff
@@ -0,0 +1,57 @@
+from keras_hub.src.api_export import keras_hub_export
+from keras_hub.src.models.qwen.qwen_backbone import Qwen2Backbone
+from keras_hub.src.tokenizers.byte_pair_tokenizer import BytePairTokenizer
+
+
+@keras_hub_export(
+    [
+        "keras_hub.tokenizers.Qwen2Tokenizer",
+        "keras_hub.models.Qwen2Tokenizer",
+    ]
+)
+class Qwen2Tokenizer(BytePairTokenizer):
+    backbone_cls = Qwen2Backbone
+
+    def __init__(
+        self,
+        vocabulary=None,
+        merges=None,
+        bos_token=None,
+        eos_token="<|endoftext|>",
+        misc_special_tokens=set(),
+        **kwargs,
+    ):
+        # Initialize special tokens set
+        special_tokens = set()
+
+        # Add BOS token if provided
+        if bos_token is not None:
+            self._add_special_token(bos_token, "start_token")
+            special_tokens.add(bos_token)
+            misc_special_tokens -= {bos_token}
+
+        # Add EOS token
+        self._add_special_token(eos_token, "end_token")
+        special_tokens.add(eos_token)
+        misc_special_tokens -= {eos_token}
+
+        # Add misc special tokens
+        for i, token in enumerate(misc_special_tokens):
```

Comment:
What are these used for? I don't see these used anywhere. A lot of tokenizers have reserved and unused tokens (e.g. for bert the first thousand I think), we don't generally give them special treatment.

Reply:
I just followed llama3 tokenizer!